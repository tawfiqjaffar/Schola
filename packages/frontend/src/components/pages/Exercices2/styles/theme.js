import gray from "@material-ui/core/colors/grey";
import lightGreen from "@material-ui/core/colors/lightGreen";

import { createMuiTheme } from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            light: lightGreen[200],
            main: lightGreen[400],
            dark: lightGreen[600],
            contrastText: "#fff"
        },
        secondary: {
            light: gray[200],
            main: gray[500],
            dark: gray[700],
            contrastText: "#fff"
        }
    }
});

export default theme;
