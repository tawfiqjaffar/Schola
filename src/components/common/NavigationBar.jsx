import React from 'react';
import {
  AppBar, Button, Link, makeStyles, Toolbar, Typography,
} from '@material-ui/core';
import logo from '../../assets/logo.png';

const useStyles = makeStyles((theme) => ({
  root: {
  },
  homeButton: {
    marginRight: theme.spacing(1),
    color: theme.palette.text.secondary,
  },
  logo: {
    width: 50,
    height: 50,
    marginRight: theme.spacing(1),
  },
  logoContainer: {
  },
  title: {
    flexGrow: 1,
    color: theme.palette.secondary.contrastText,
  },
  menuButton: {
    padding: theme.spacing(1),
    marginRight: theme.spacing(1),
    color: theme.palette.secondary.contrastText,
    '&:hover': {
      backgroundColor: theme.palette.primary.dark,
    },
  },
  buttonRegister: {
    backgroundColor: theme.palette.secondary.main,
  },
  buttonLogin: {
    backgroundColor: theme.palette.secondary.light,
  },
}));

export default function NavigationBar() {
  const classes = useStyles();

  return (
    <AppBar position="sticky" className={classes.root}>
      <Toolbar>
        <Link
          href="/"
        >
          <img src={logo} className={classes.logo} alt="Schola logo" />
        </Link>
        <Typography
          className={classes.title}
          variant="h5"
        >
          Schola
        </Typography>
        <Button
          className={[classes.menuButton, classes.buttonLogin]}
          variant="contained"
        >
          <Typography>Log in</Typography>
        </Button>
        <Button
          className={[classes.menuButton, classes.buttonRegister]}
          variant="contained"
        >
          <Typography>Register</Typography>
        </Button>
      </Toolbar>
    </AppBar>
  );
}
