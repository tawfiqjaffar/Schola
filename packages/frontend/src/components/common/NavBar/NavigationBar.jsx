import React from 'react';
// import { useSelector } from 'react-redux';
import {
  AppBar,
  Button,
  Link,
  makeStyles,
  Toolbar,
  Typography,
  ListItemText,
  Menu,
  MenuItem,
  IconButton,
} from '@material-ui/core';
import Cookies from 'js-cookie';
import { useHistory } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import MenuIcon from '@material-ui/icons/Menu';
import logo from '../../../assets/logo.png';
import './NavBar.css';
import navRoutes from './NavBarRoute';

const useStyles = makeStyles((theme) => ({
  root: {
  },
  homeButton: {
    marginRight: theme.spacing(1),
    color: theme.palette.text.secondary,
  },
  logo: {
    width: 50,
    height: 50,
    marginRight: theme.spacing(1),
  },
  logoContainer: {},
  menuButton: {
    padding: theme.spacing(1),
    marginRight: theme.spacing(1),
    color: theme.palette.secondary.contrastText,
    '&:hover': {
      backgroundColor: theme.palette.primary.dark,
    },
  },
  buttonRegister: {
    backgroundColor: theme.palette.secondary.main,
  },
  buttonLogin: {
    backgroundColor: theme.palette.secondary.light,
  },
}));

function LoginLogoutDisplay(props) {
  const { isLogged, redirection, disconnect } = props;
  if (!isLogged) {
    return <LoginRegister redirection={redirection} />;
  }
  return <Logout disconnect={disconnect} redirection={redirection} />;
}

const StyledMenu = withStyles()((props) => (
  <Menu
    elevation={3}
    getContentAnchorEl={null}
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center',
    }}
    transformOrigin={{
      vertical: 'top',
      horizontal: 'center',
    }}
    // eslint-disable-next-line react/jsx-props-no-spreading
    {...props}
  />
));

const StyledMenuItem = withStyles((theme) => ({
  root: {
    '&:focus': {
      backgroundColor: '#70c6c7',
      '& .MuiListItemIcon-root, & .MuiListItemText-primary': {
        color: theme.palette.common.white,
      },
    },
  },
}))(MenuItem);

function Logout(props) {
  const { disconnect, redirection } = props;
  const [anchorEl, setAnchorEl] = React.useState(null);
  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <div>
      <IconButton onClick={handleClick} className="ButtonRight">
        <MenuIcon style={{ width: '50px', height: '50px' }} />
      </IconButton>
      <StyledMenu
        id="customized-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        {navRoutes.map((el) =>
        // if (!el.access.includes(myRole)) {
        //   return (<div />);
        // }

          (
            <StyledMenuItem
              onClick={() => {
                redirection(el.route);
              }}
            >
              <ListItemText primary={el.label} />
            </StyledMenuItem>
          ))}
        <StyledMenuItem onClick={() => disconnect()}>
          <ListItemText primary="Disconnect" />
        </StyledMenuItem>
      </StyledMenu>
    </div>
  );
}

function LoginRegister(props) {
  const { redirection } = props;
  const classes = useStyles();
  return (
    <div className="ButtonRight">
      <Button
        className={[classes.menuButton, classes.buttonLogin].join(' ')}
        variant="contained"
        onClick={() => redirection('/login')}
      >
        <Typography>Log in</Typography>
      </Button>
      <Button
        className={[classes.menuButton, classes.buttonRegister].join(' ')}
        variant="contained"
      >
        <Typography>Register</Typography>
      </Button>
    </div>
  );
}

export default function NavigationBar() {
  const classes = useStyles();
  // const user = useSelector((state) => state.user);
  const history = useHistory();
  const redirectTo = (path) => {
    history.push(path);
  };
  const redirectToLandingPage = () => {
    history.push('/');
    Cookies.remove('accessToken');
    window.location.reload();
  };
  let isLogged = true;
  if (!Cookies.get('accessToken')) isLogged = false;

  return (
    <AppBar position="sticky" className={classes.root}>
      <Toolbar className="Container100">
        <div className="wrapper">
          <Link href="/" className="CenterImg">
            <img src={logo} className={classes.logo} alt="Schola logo" />
          </Link>
          <Typography className="title" variant="h5">
            Schola
          </Typography>
        </div>
        <LoginLogoutDisplay
          isLogged={isLogged}
          disconnect={redirectToLandingPage}
          redirection={redirectTo}
        />
      </Toolbar>
    </AppBar>
  );
}

LoginRegister.propTypes = {
  redirection: PropTypes.func.isRequired,
};

Logout.propTypes = {
  disconnect: PropTypes.func.isRequired,
  redirection: PropTypes.func.isRequired,
};

LoginLogoutDisplay.propTypes = {
  disconnect: PropTypes.func.isRequired,
  redirection: PropTypes.func.isRequired,
  isLogged: PropTypes.bool.isRequired,
};
