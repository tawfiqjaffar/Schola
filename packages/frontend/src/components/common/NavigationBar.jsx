import React from 'react';
import {
  AppBar,
  Button,
  Link,
  makeStyles,
  Toolbar,
  Typography,
  ListItemText,
  Menu,
  MenuItem,
  IconButton,
} from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import MenuIcon from '@material-ui/icons/Menu';
import logo from '../../assets/logo.png';

const useStyles = makeStyles((theme) => ({
  root: {},
  homeButton: {
    marginRight: theme.spacing(1),
    color: theme.palette.text.secondary,
  },
  logo: {
    width: 50,
    height: 50,
    marginRight: theme.spacing(1),
  },
  logoContainer: {},
  title: {
    flexGrow: 1,
    color: theme.palette.secondary.contrastText,
  },
  menuButton: {
    padding: theme.spacing(1),
    marginRight: theme.spacing(1),
    color: theme.palette.secondary.contrastText,
    '&:hover': {
      backgroundColor: theme.palette.primary.dark,
    },
  },
  buttonRegister: {
    backgroundColor: theme.palette.secondary.main,
  },
  buttonLogin: {
    backgroundColor: theme.palette.secondary.light,
  },
}));

function LoginLogoutDisplay(props) {
  const { isLogged, redirection, disconnect } = props;
  console.log(isLogged);
  if (!isLogged) {
    return <LoginRegister redirection={redirection} />;
  }
  return <Logout disconnect={disconnect} redirection={redirection} />;
}

const StyledMenu = withStyles()((props) => (
  <Menu
    elevation={3}
    getContentAnchorEl={null}
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center',
    }}
    transformOrigin={{
      vertical: 'top',
      horizontal: 'center',
    }}
    // eslint-disable-next-line react/jsx-props-no-spreading
    {...props}
  />
));

const StyledMenuItem = withStyles((theme) => ({
  root: {
    '&:focus': {
      backgroundColor: '#2a8081',
      '& .MuiListItemIcon-root, & .MuiListItemText-primary': {
        color: theme.palette.common.white,
      },
    },
  },
}))(MenuItem);

function Logout(props) {
  const { disconnect, redirection } = props;
  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <div>
      <IconButton onClick={handleClick}>
        <MenuIcon />
      </IconButton>
      <StyledMenu
        id="customized-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <StyledMenuItem
          onClick={() => {
            redirection('/home');
          }}
        >
          <ListItemText primary="Home" />
        </StyledMenuItem>
        <StyledMenuItem
          onClick={() => {
            redirection('/schedule');
          }}
        >
          <ListItemText primary="Emploi du temps" />
        </StyledMenuItem>
        <StyledMenuItem
          onClick={() => {
            redirection('/canteen');
          }}
        >
          <ListItemText primary="Cantine" />
        </StyledMenuItem>
        <StyledMenuItem
          onClick={() => {
            redirection('/quizz');
          }}
        >
          <ListItemText primary="Quizz" />
        </StyledMenuItem>
        <StyledMenuItem
          onClick={() => {
            redirection('/rollcall');
          }}
        >
          <ListItemText primary="RollCall" />
        </StyledMenuItem>
        <StyledMenuItem onClick={() => disconnect()}>
          <ListItemText primary="Disconnect" />
        </StyledMenuItem>
      </StyledMenu>
    </div>
  );
}

function LoginRegister(props) {
  const { redirection } = props;
  const classes = useStyles();
  return (
    <div>
      <Button
        className={[classes.menuButton, classes.buttonLogin].join(' ')}
        variant="contained"
        onClick={() => redirection('/login')}
      >
        <Typography>Log in</Typography>
      </Button>
      <Button
        className={[classes.menuButton, classes.buttonRegister].join(' ')}
        variant="contained"
      >
        <Typography>Register</Typography>
      </Button>
    </div>
  );
}

export default function NavigationBar() {
  const classes = useStyles();
  const history = useHistory();
  const redirectTo = (path) => {
    history.push(path);
  };
  const redirectToLandingPage = () => {
    history.push('/');
    sessionStorage.removeItem('token');
    window.location.reload();
  };
  let isLogged = true;
  if (!sessionStorage.getItem('token')) isLogged = false;

  return (
    <AppBar position="sticky" className={classes.root}>
      <Toolbar>
        <Link href="/">
          <img src={logo} className={classes.logo} alt="Schola logo" />
        </Link>
        <Typography className={classes.title} variant="h5">
          Schola
        </Typography>
        <LoginLogoutDisplay
          isLogged={isLogged}
          disconnect={redirectToLandingPage}
          redirection={redirectTo}
        />
      </Toolbar>
    </AppBar>
  );
}

LoginRegister.propTypes = {
  redirection: PropTypes.func.isRequired,
};

Logout.propTypes = {
  disconnect: PropTypes.func.isRequired,
  redirection: PropTypes.func.isRequired,
};

LoginLogoutDisplay.propTypes = {
  disconnect: PropTypes.func.isRequired,
  redirection: PropTypes.func.isRequired,
  isLogged: PropTypes.bool.isRequired,
};
