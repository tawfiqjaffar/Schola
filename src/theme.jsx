import { createMuiTheme } from '@material-ui/core/styles';

const palette = {
  primary: { main: '#70C6C7', contrastText: '#212121' },
  secondary: { main: '#60c07a', contrastText: '#FAFAFA' },
  text: {
    primary: '#212121',
    secondary: '#FAFAFA',
  },
  background: {
    paper: '#70C6C7',
  },
};
const themeName = 'Downy Pastel Green Gayal';

export default createMuiTheme({ palette, themeName });
