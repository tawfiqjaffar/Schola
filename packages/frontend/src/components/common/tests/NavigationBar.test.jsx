import React from 'react';

import renderer from 'react-test-renderer';
import { ThemeProvider } from '@material-ui/core';
import { BrowserRouter } from 'react-router-dom';
import NavigationBar from '../NavigationBar';
import theme from '../../../theme';

describe('NavigationBar', () => {
  it('should render NavigationBar and match snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <NavigationBar />
        </BrowserRouter>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
