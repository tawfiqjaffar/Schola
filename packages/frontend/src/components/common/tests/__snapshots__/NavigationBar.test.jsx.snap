// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar should render NavigationBar and match snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary makeStyles-root-1 MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="/"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <img
        alt="Schola logo"
        className="makeStyles-logo-3"
        src="logo.png"
      />
    </a>
    <h5
      className="MuiTypography-root makeStyles-title-5 MuiTypography-h5"
    >
      Schola
    </h5>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-menuButton-6 makeStyles-buttonLogin-8"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Log in
        </p>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-menuButton-6 makeStyles-buttonRegister-7"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Register
        </p>
      </span>
    </button>
  </div>
</header>
`;
